Detectors/TRD/simulation/src/Digitizer.cxx

```diff
@@ -22,9 +41,402 @@ Digitizer::~Digitizer()
 
 void Digitizer::process(std::vector<o2::trd::HitType> const& hits, std::vector<o2::trd::Digit>& digits)
 {
-  // very basic implementation
-  // for every hit seen, create a digit
-  for (auto& hit : hits) {
+  // (WIP) Implementation for digitization
+
+  // Check if Geometry and if CCDB are available as they will be requiered
+  /*
+   mGeom = new TRDGeometry();
+   TRDCalibDB * calibration = new TRDCalibDB();
+   const int nTimeBins = calibration->GetNumberOfTimeBinsDCS();
+  */
+
+  // Loop over all TRD detectors
+  // Get the a hit container for all the hits in a given detector then call convertHits for a given detector (0 - 539)
+  const int kNdet = 540; // Get this from TRD Geometry
+  int totalNumberOfProcessedHits = 0;
+  LOG(INFO) << "Start of processing " << hits.size() << " hits";
+  for (int det = 0; det < kNdet; ++det) {
+    // Loop over all TRD detectors
+
+    // Jump to the next detector if the detector is
+    // switched off, not installed, etc
+    /*
+    if (calibration->IsChamberNoData(det)) {
+      continue;
+    }
+    if (!mGeo->ChamberInGeometry(det)) {
+      continue
+    }
+    */
+
```

Comment:
I would clear the container here, before you fill it.

Reply:
Thanks, I missed that point after moving the container as a member of the class.